### DISCUSSION TITLE
Zowe zos-jobs submit --wait-for-output option

### DISCUSSION THREAD

**Discussion by sureshbabukannan** [zowe/zowe-cli#1795] [Q&A]
Dear Zowe commumity memeber, 

This is related to changelle in knowing the job completion when the job output is sent to message Class other than default JES3 job output queue . We are looking for some configuration to zos-jobs where we could find the jobs status in other third pary output management systems in this case $AVRS. 

You could find the details on the below

**Problem Statement:**

Mainframe Devops process are integrated using github, IBM dependency based Build (DBB) , Rocket Open App Dev for z tools, Zowe frame work. The automated Devops processes run as codefresh pipelines orcestrates these tools. As part of automation there is requirement to submit jobs to mainframe using **zowe CLi command zos-jobs submit**  There should be a mechanism to return the status or completion code of the job back to codefresh pipeline where the job was intiated.

**Challenge in getting the solution**
zowe zos-jobs submit command does provide a option --wait-for-output. This option lets the command synchoronously wait for the job completed and to get completed and sent to JES3 OUTPUT queue.  In our mainframe shop the job output is directed to third party output management system called $AVRS on mainframe. within the split seconds of job output reaching the JES3 output queue, it is again redirected to $AVRS output management subsystem. This means zos-jobs submit command could not intercept the job on JES3 output queue and fails with HTTP 400 status code.

**Question to the zowe community**
Can the zos-jobs submit command made to wait on message class other than default JES3 output queue message class . This will allow the command to get the job return code from the Job output to any of the output queue , not just the JES3 output queue.

**Comment by balhar-jakub** [comment]
@adam-wolfe Can someone from the CLI squad take a look?
